diff -urN emsdk-pure/upstream/emscripten/emcc.py emsdk/upstream/emscripten/emcc.py
--- emsdk-pure/upstream/emscripten/emcc.py	2022-08-02 14:52:59.442877733 +0000
+++ emsdk/upstream/emscripten/emcc.py	2022-08-02 14:52:18.131057940 +0000
@@ -624,6 +624,8 @@
   cmd = config.JS_ENGINE
   if settings.WASM_BIGINT:
     cmd.append('--experimental-wasm-bigint')
+  if settings.USE_PTHREADS:
+    cmd.append('--experimental-wasm-threads')
   cmd = shared.shlex_join(cmd)
   if not os.path.isabs(config.JS_ENGINE[0]):
     # TODO: use whereis etc. And how about non-*NIX?
