--- emsdk1/upstream/emscripten/emcc.py	2022-03-24 19:45:37.000000000 +0000
+++ emsdk2/upstream/emscripten/emcc.py	2022-03-28 19:42:22.309947641 +0000
@@ -605,9 +605,10 @@
 def make_js_executable(script):
   src = read_file(script)
   cmd = shared.shlex_join(config.JS_ENGINE)
-  if not os.path.isabs(config.JS_ENGINE[0]):
-    # TODO: use whereis etc. And how about non-*NIX?
-    cmd = '/usr/bin/env -S ' + cmd
+  #if not os.path.isabs(config.JS_ENGINE[0]):
+  #  # TODO: use whereis etc. And how about non-*NIX?
+  cmd = '/usr/bin/env -S ' + cmd
+  cmd = cmd + ' --experimental-wasm-threads'
   logger.debug('adding `#!` to JavaScript file: %s' % cmd)
   # add shebang
   with open(script, 'w') as f:
