--- emsdk-orig/upstream/emscripten/system/lib/wasmfs/syscalls.cpp	2022-07-29 17:22:28.000000000 +0000
+++ emsdk/upstream/emscripten/system/lib/wasmfs/syscalls.cpp	2022-08-06 05:05:17.014502697 +0000
@@ -1419,7 +1419,7 @@
       static_assert(F_SETLK == F_SETLK64);
       static_assert(F_SETLKW == F_SETLKW64);
       // Always error for now, until we implement byte-range locks.
-      return -EACCES;
+      return 0; //-EACCES;
     }
     case F_GETOWN_EX:
     case F_SETOWN:
