diff -urN emsdk/upstream/emscripten/emcc.py emsdk-fixed/upstream/emscripten/emcc.py
--- emsdk/upstream/emscripten/emcc.py	2022-11-07 20:04:28.000000000 +0000
+++ emsdk-fixed/upstream/emscripten/emcc.py	2022-11-11 02:48:00.766940527 +0000
@@ -632,6 +632,8 @@
   cmd = config.JS_ENGINES[0]
   if settings.WASM_BIGINT:
     cmd += shared.node_bigint_flags()
+  if settings.USE_PTHREADS:
+    cmd.append('--experimental-wasm-threads')
   cmd = shared.shlex_join(cmd)
   if not os.path.isabs(cmd[0]):
     # TODO: use whereis etc. And how about non-*NIX?
